ethereum_history_api/oracles/src/script/noir_fixtures/transaction.ts

```diff
@@ -0,0 +1,35 @@
+import { GetTransactionReturnType } from 'viem';
+import { encodeHexString, encodeNullable, formatArray } from '../../noir/noir_js/encode.js';
+import { ZERO_PAD_VALUE } from '../../noir/oracles/common/const.js';
+import { encodeAddress, encodeField } from '../../noir/oracles/common/encode.js';
+import { MAX_TRANSACTION_DATA_SIZE } from '../../noir/oracles/transactionOracle/encode.js';
+import { padArray } from '../../util/array.js';
+
+export function createTransactionFixture(tx: GetTransactionReturnType): string {
+  const nonce = encodeField(tx.nonce);
+  const gasPrice = encodeNullable(tx.gasPrice !== undefined ? encodeField(tx.gasPrice) : null);
```

Comment:
Why convert something that's maybe undefined to something that's maybe null? Why not make `encodeOptionalField(bignum|undefined) -> string`

Reply:
Changed to `encodeOptional`. Sometimes missing values are null like `tx.to` or `undefined` like `tx.gasPrice` so we need to handle both options. Moreover I want to make optional not only Field but also other types like arrays - so I need to separate encoding field from optional.